bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+import Keyring from '@polkadot/keyring';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  getChainflipApi,
+  observeEvent,
+  lpMutex,
+  chainFromAsset,
+  shortChainFromAsset,
+  amountToFineAmount,
+  assetDecimals,
+  Event,
+  ingressEgressPalletForChain,
+  submitChainflipExtrinsic,
+  calculateFeeWithBps,
+  amountToFineAmountBigInt,
+  newAddress,
+} from './utils';
+import { send } from './send';
+import { provideLiquidity } from './provide_liquidity';
+import { requestNewSwap } from './perform_swap';
+import { createBoostPools } from './setup_boost_pools';
+
+const keyring = new Keyring({ type: 'sr25519' });
+keyring.setSS58Format(2112);
+
+async function lpSubmitExtrinsic(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  extrinsic: any,
+  keyringPair: KeyringPair,
+  errorOnFail: boolean = true,
+) {
+  let extrinsicResult;
+  await lpMutex.runExclusive(async () => {
```

Comment:
It seems strange to me that we hardcode `lpMutex` here, while a different account can passed in as a parameter. We should either hardcode both the acount and the mutex, or provide a way to set a custom mutex too.

Reply:
That's why i called the function `lpSubmitExtrinsic`. Because it uses the LP mutex. Seemed ok to use the same mutex for different LP accounts. It's not perfect. If we end up using it somewhere else, I would move it to utils and make the mutex a param. Unless you think its worth doing that now?

Reply:
The main point of the function is just to make it easier to get the extrinsic result. Because of the mutex, I am unable to get the result directly and must store it in a variable first. This makes it messy.

Reply:
I don't really mind the way it is written now. It is a little strange, but it does work (I was initially concerned about using a wrong mutex, but you are right that different accounts can share the same mutex).

I do like Kyle's suggestion re creating a class (which I imagine would take a mutex and a keyringPair in its constructor). Sounds like this might then be reused in other places too. Unless we are in the rush to merge this, it might be worth doing it here?